src/omefiles.cpp

```diff
@@ -26,67 +26,139 @@
+    .def(py::init<uint32_t, uint32_t, uint32_t, const std::string &>())
+    .def_readonly("major", &ome::files::Version::major)
+    .def_readonly("minor", &ome::files::Version::minor)
+    .def_readonly("patch", &ome::files::Version::patch)
+    .def_readonly("extra", &ome::files::Version::extra);
+  m.attr("version_info") = ome::files::release_version;
+  py::class_<OMETIFFReader>(m, "OMETIFFReader")
+    .def("__init__", [](OMETIFFReader &r) {
+	new (&r) OMETIFFReader();
+	std::shared_ptr<ome::xml::meta::MetadataStore> store(
+          std::make_shared<ome::xml::meta::OMEXMLMetadata>()
+        );
+	r.setMetadataStore(store);
+      })
+    .def("set_id", [](OMETIFFReader &r, std::string id) {
+	r.setId(id);
+      }, "Set the current file name.")
+    .def("set_series", [](OMETIFFReader &r, size_t series) {
+        r.setSeries(series);
+      }, "Set the active series.")
+    .def("get_series", [](OMETIFFReader &r) {
+        return r.getSeries();
+      }, "Get the active series.")
+    .def("get_image_count", [](OMETIFFReader &r) {
+	return r.getImageCount();
+      }, "Get the number of image planes in the current series.")
+    .def("get_series_count", [](OMETIFFReader &r) {
+        return r.getSeriesCount();
+      }, "Get the number of image series.")
+    .def("get_size_x", [](OMETIFFReader &r) {
+	return r.getSizeX();
+      }, "Get the size of the X dimension.")
+    .def("get_size_y", [](OMETIFFReader &r) {
+        return r.getSizeY();
+      }, "Get the size of the Y dimension.")
+    .def("get_size_z", [](OMETIFFReader &r) {
+        return r.getSizeZ();
+      }, "Get the size of the Z dimension.")
+    .def("get_size_t", [](OMETIFFReader &r) {
+        return r.getSizeT();
+      }, "Get the size of the T dimension.")
+    .def("get_size_c", [](OMETIFFReader &r) {
+        return r.getSizeC();
+      }, "Get the size of the C dimension.")
+    .def("get_dimension_order", [](OMETIFFReader &r) {
+        return r.getDimensionOrder();
+      }, "Get a five-character string representing the order in which "
+      "planes will be returned.")
+    .def("get_index", [](OMETIFFReader &r, size_t z, size_t c, size_t t) {
+	return r.getIndex(z, c, t);
+      }, "Get the plane index corresponding to the given ZCT coordinates.")
+    .def("get_zct_coords", [](OMETIFFReader &r, size_t index) {
+	return r.getZCTCoords(index);
+      }, "Get the ZCT coordinates for the given plane index.")
+    .def("get_rgb_channel_count", [](OMETIFFReader &r, size_t channel) {
+	return r.getRGBChannelCount(channel);
+      }, "Get the number of sub-channels for the given channel.")
+    .def("is_interleaved",
+	 (bool (OMETIFFReader::*)(size_t) const) &OMETIFFReader::isInterleaved,
+	 "Whether or not the given channel is interleaved")
+    .def("is_interleaved",
+	 (bool (OMETIFFReader::*)() const) &OMETIFFReader::isInterleaved,
+	 "Whether or not the channels are interleaved")
+    .def("get_pixel_type", [](OMETIFFReader &r) {
```

Comment:
We could do that in a future PR, better not change the Python interface too much in this one.